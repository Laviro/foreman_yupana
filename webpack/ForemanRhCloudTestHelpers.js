export const inventoryStateWrapper = innerState => ({
  ForemanRhCloud: {
    inventoryUpload: { ...innerState },
  },
});
