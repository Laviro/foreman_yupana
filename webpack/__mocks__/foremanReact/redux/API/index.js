export const get = data => ({ type: 'get-some-type', ...data });
export const put = data => ({ type: 'put-some-type', ...data });
export const post = data => ({ type: 'post-some-type', ...data });
export const patch = data => ({ type: 'patch-some-type', ...data });

export const API = {
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  patch: jest.fn(),
};
