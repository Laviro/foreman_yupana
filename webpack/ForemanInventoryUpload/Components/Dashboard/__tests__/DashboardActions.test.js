import { testActionSnapshotWithFixtures } from 'react-redux-test-utils';
import API from 'foremanReact/API';
import {
  startPolling,
  stopPolling,
  fetchLogs,
  setActiveTab,
  downloadReports,
  toggleFullScreen,
} from '../DashboardActions';
import {
  pollingProcessID,
  serverMock,
  activeTab,
  accountID,
} from '../Dashboard.fixtures';
import { rhCloudStateWrapper } from '../../../../ForemanRhCloudTestHelpers';

jest.mock('foremanReact/API');
API.get.mockImplementation(() => serverMock);

const runWithGetState = (state, action, params) => dispatch => {
  const getState = () => rhCloudStateWrapper({ dashboard: state });
  action(params)(dispatch, getState);
};

const fixtures = {
  'should startPolling': () => startPolling(accountID, pollingProcessID),
  'should fetchLogs': () =>
    runWithGetState({ activeTab: 'uploads' }, fetchLogs, accountID),
  'should stopPolling': () => stopPolling(accountID, pollingProcessID),
  'should setActiveTab': () => setActiveTab(accountID, activeTab),
  'should downloadReports': () => downloadReports(accountID),
  'should toggleFullScreen': () =>
    runWithGetState({ activeTab: 'reports' }, toggleFullScreen, accountID),
};

describe('Dashboard actions', () => testActionSnapshotWithFixtures(fixtures));
