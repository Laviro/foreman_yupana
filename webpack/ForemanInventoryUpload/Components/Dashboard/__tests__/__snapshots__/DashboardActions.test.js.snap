// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard actions should downloadReports 1`] = `
Object {
  "payload": Object {
    "accountID": "some-account-ID",
  },
  "type": "INVENTORY_REPORTS_DOWNLOAD",
}
`;

exports[`Dashboard actions should fetchLogs 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "accountID": "some-account-ID",
        "activeTab": undefined,
        "logs": Array [
          "some-logs",
          "some-logs",
        ],
      },
      "type": "INVENTORY_POLLING",
    },
  ],
]
`;

exports[`Dashboard actions should restartProcess 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "accountID": "some-account-ID",
      },
      "type": "INVENTORY_PROCESS_RESTART",
    },
  ],
]
`;

exports[`Dashboard actions should setActiveTab 1`] = `
Object {
  "payload": Object {
    "accountID": "some-account-ID",
    "activeTab": "uploads",
  },
  "type": "INVENTORY_TAB_CHANGED",
}
`;

exports[`Dashboard actions should startPolling 1`] = `
Object {
  "payload": Object {
    "accountID": "some-account-ID",
    "pollingProcessID": 1,
  },
  "type": "INVENTORY_POLLING_START",
}
`;

exports[`Dashboard actions should stopPolling 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "accountID": "some-account-ID",
      },
      "type": "INVENTORY_POLLING_STOP",
    },
  ],
]
`;
